### ISSUE TITLE
Resolve Pitest Issues - ConfigurationLoader

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#9325]
Child issue of #7797

Blocker for #9146
Example of failed [build](https://app.circleci.com/pipelines/github/pbludov/checkstyle/938/workflows/54a07833-3b2d-4687-837d-7dace6ad168e/jobs/6267).
```bash
+ConfigurationLoader.java.html:<td class='uncovered'><pre><span  class=''>                        throw new SAXException(</span></pre></td></tr>
 ConfigurationLoader.java.html:<td class='uncovered'><pre><span  class=''>                    catch (final CheckstyleException ex) {</span></pre></td></tr>
 ConfigurationLoader.java.html:<td class='uncovered'><pre><span  class='survived'>                                        + recentModule.getName(), ex);</span></pre></td></tr>
-ConfigurationLoader.java.html:<td class='uncovered'><pre><span  class='survived'>                        throw new SAXException(</span></pre></td></tr>
```
The code is
```java
                if (containsAttribute(recentModule, SEVERITY)) {
                    try {
                        final String severity = recentModule.getAttribute(SEVERITY);
                        level = SeverityLevel.getInstance(severity);
                    }
                    catch (final CheckstyleException ex) {
                        // -@cs[IllegalInstantiation] SAXException is in the overridden
                        // method signature
                        throw new SAXException(
                                "Problem during accessing '" + SEVERITY + "' attribute for "
                                        + recentModule.getName(), ex);
                    }
                }
```
The `catch` block has no chance to execute, since it is checked already that this attribute exists.
This can be solved by changing the signature of `containsAttribute` from `boolean` to Optional<String>`, ie return the optional value for the attribute instead of the presence flag for the attribute.

**Comment by rnveach** [comment]
> The catch block has no chance to execute

So your saying there is no code coverage on these blocks?

> This can be solved by changing the signature of containsAttribute from boolean to Optional`

I am not seeing how this is connecting to the try/catch. The if statement has no bearing on if, what I assume is, `recentModule` throws an exception or not.

**Comment by pbludov** [comment]
> I am not seeing how this is connecting to the try/catch. The if statement has no bearing on if, what I assume is, `recentModule` throws an exception or not.

`recentModule`  will never throw such exception. ` if (containsAttribute(recentModule, SEVERITY))` guarantees that such attribute is defined for the module.

Another solution is to remove the `containsAttribute`:
```java
                // get severity attribute if it exists
                SeverityLevel level;
                try {
                    final String severity = recentModule.getAttribute(SEVERITY);
                    level = SeverityLevel.getInstance(severity);
                }
                catch (final CheckstyleException ignored) {
                    level = null;
                }
```
Or we can add overload to `getAttribute`:
```java
public interface Configuration extends Serializable {
    // ...
    String getAttribute(String name) throws CheckstyleException;

    String getAttribute(String name, String defaultValue); // this method never throws an exception 
    // ...
}
```

**Comment by pbludov** [comment]
Fixed by #10932
